### PR TITLE
rt.sh: ecflow compile jobs, update stampede config, remove cheyenne.pgi, check for GNU compiler version

### PR DESCRIPTION
**Pull request by climbfuji**
This PR contains the following updates:
- update stampede config, remove cheyenne.pgi config, bugfixes for Python3 on stampede (from @climbfuji)
- check version of GNU compiler #134 (from @edwardhartnett)
- submit compile jobs with ecflow to batch system instead of running them on frontend nodes (from @DusanJovic-NOAA)

Associated PRs:
https://github.com/NCAR/ccpp-framework/pull/295
https://github.com/NOAA-EMC/fv3atm/pull/117
https://github.com/NOAA-EMC/NEMS/pull/63
https://github.com/ufs-community/ufs-weather-model/pull/139

For regression testing information, see below.

### CODE REVIEW

**Comment by climbfuji** [comment]
Manual testing on macOS with clang+gfortran and gcc+gfortran, Python 2.7 and Python 3.7: code compiles for both `compile.sh` and `compile_cmake.sh`, with IPD and CCPP.

**Comment by climbfuji** [comment]
Regression testing on hera.intel (using `rt.sh -f -e`) and orion.intel (using `rt.sh -l rt_orion.conf -e`): all tests passed, regression test logs updated in the PR.

**Comment by climbfuji** [comment]
Regression testing on wcoss_cray and wcoss_dell_p3: all tests pass, regression test logs updated in the PR.

**Comment by climbfuji** [comment]
Manual testing on stampede: create new baseline using `rt.sh -l rt_stampede_cheyenne.conf -c`, then rsync to RTPWD (user-specific, since we do not have a project directory on stampede yet), then verify against it using `rt.sh -l rt_stampede_cheyenne.conf`: all tests pass.

[rt_stampede_intel_create.log](https://github.com/ufs-community/ufs-weather-model/files/4713410/rt_stampede_intel_create.log)
[rt_stampede_intel_verify.log](https://github.com/ufs-community/ufs-weather-model/files/4713411/rt_stampede_intel_verify.log)

**Comment by climbfuji** [comment]
Also tested compiling and running on cheyenne.intel and cheyenne.gnu, successful. These PRs are ready to merge.

File: tests/compile_cmake.sh [hunk 1]
```diff
@@ -42,23 +32,22 @@ else
...
+BUILD_DIR=$(pwd)/build_${BUILD_NAME}
 
 # ----------------------------------------------------------------------
 # Make sure we have reasonable number of threads.
 
 if [[ $MACHINE_ID == cheyenne.* ]] ; then
     MAKE_THREADS=${MAKE_THREADS:-3}
 elif [[ $MACHINE_ID == wcoss_dell_p3 ]] ; then
-    MAKE_THREADS=${MAKE_THREADS:-4}
+    MAKE_THREADS=${MAKE_THREADS:-1}
```
Comment (junwang-noaa) [inline tests/compile_cmake.sh:44]: May I ask why the MAKE_THREADS is changed to one? Is it the reason the model compile is taking longer time now?

Comment (climbfuji) [reply tests/compile_cmake.sh:44]: This change is from @DusanJovic-NOAA .

Comment (DusanJovic-NOAA) [reply tests/compile_cmake.sh:44]: Yes. Compilation now runs on compute nodes and on WCOSS Dell is using 'dev_transfer' queue which allows only one task. So each COMPILE step is running little slower but on the other hand we can simultaneously submit up to 10 COMPILE steps.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**